bpf/bpf_lxc.c

```diff
@@ -1778,8 +1798,10 @@ ipv4_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label, enum ct_status
 			return ret2;
 
 		ct_state_new.dsr = dsr;
-		if (ret == CT_REOPENED && ct_state->dsr != dsr)
+		if ((ret == CT_REOPENED || ret == CT_ESTABLISHED) && ct_state->dsr != dsr) {
 			ct_update4_dsr(get_ct_map4(tuple), tuple, dsr);
+			ct_update_nodeport(get_ct_map4(tuple), tuple, !dsr);
```

Comment:
As commented above, this call is not needed.

Reply:
Just to be clear, do you mean that the nodeport flag is irrelevant to the DSR case(the nodeport flag is for the SNAT case), therefore we don't need to touch it here?